stdlib/public/Concurrency/Task.cpp

```diff
@@ -1671,7 +1671,8 @@ static void swift_task_asyncMainDrainQueueImpl() {
   }
 #endif
 
-    dispatch_main();
+  dispatch_main();
+  swift_unreachable("Returned from dispatch_main()");
```

Comment:
I think that `swift_Concurrency_fatalError` might be more appropriate here.

Reply:
The other ifdefs end with swift_unreachable. Should I change those with fatal errors too?